lib/ansible/modules/storage/netapp/na_ontap_net_vlan.py

```diff
@@ -0,0 +1,198 @@
+        self.vlanid = p['vlanid']
+        self.node = p['node']
+        self.interface_name = p['interface_name']
+        self.gvrp_enabled = p['gvrp_enabled']
+
+        if HAS_NETAPP_LIB is False:
+            self.module.fail_json(msg="the python NetApp-Lib module is required")
+        else:
+            self.server = netapp_utils.setup_ontap_zapi(module=self.module)
+        return
+
+    def create_vlan(self):
+        """
+        Creates a new vlan
+        """
+        vlan_obj = netapp_utils.zapi.NaElement("net-vlan-create")
+        vlan_info = self.create_vlan_info()
+
+        vlan_obj.add_child_elem(vlan_info)
+        self.server.invoke_successfully(vlan_obj, True)
+
+    def delete_vlan(self):
+        """
+        Deletes a vland
+        """
+        vlan_obj = netapp_utils.zapi.NaElement("net-vlan-delete")
+        vlan_info = self.create_vlan_info()
+
+        vlan_obj.add_child_elem(vlan_info)
+        self.server.invoke_successfully(vlan_obj, True)
+
+    def does_vlan_exits(self):
+        """
+        Checks to see if a vlan already exists or not
+        :return: Returns True if the vlan exists, false if it dosn't
+        """
+        vlan_obj = netapp_utils.zapi.NaElement("net-vlan-get")
+        vlan_obj.add_new_child("interface-name", self.parent_interface + "-" + self.vlanid)
+        vlan_obj.add_new_child("node", self.node)
+        try:
+            result = self.server.invoke_successfully(vlan_obj, True)
+            result.get_child_by_name("attributes").get_child_by_name("vlan-info").get_child_by_name("interface-name")
+        except netapp_utils.zapi.NaApiError:
+            return False
+        return True
+
+    def create_vlan_info(self):
+        """
+        Create a vlan_info object to be used in a create/delete
+        :return:
+        """
+        vlan_info = netapp_utils.zapi.NaElement("vlan-info")
+
+        #  set up the vlan_info object:
+        vlan_info.add_new_child("parent-interface", self.parent_interface)
+        vlan_info.add_new_child("vlanid", self.vlanid)
+        #  add the optional line if they exist.
+        if self.node:
+            vlan_info.add_new_child("node", self.node)
+        if self.interface_name:
+            vlan_info.add_new_child("interface-name", self.interface_name)
+        if self.gvrp_enabled:
+            vlan_info.add_new_child("gvrp-enabled", self.gvrp_enabled)
+        return vlan_info
+
+    def apply(self):
+        """
+        check the option in the playbook to see what needs to be done
+        :return:
+        """
+        changed = False
+        result = None
+        results = netapp_utils.get_cserver(self.server)
+        cserver = netapp_utils.setup_ontap_zapi(module=self.module, vserver=results)
+        netapp_utils.ems_log_event("na_ontap_net_vlan", cserver)
+        existing_vlan = self.does_vlan_exits()
+        if existing_vlan:
+            if self.state == 'absent':  # delete
+                changed = True
+        else:
```

Comment:
should probably be validating whichever components aren't part of the vlan's "primary key" (seems like parent interface/node/interface_name, but not sure off the top of my head) and either allow them to be modified or at least warn/error if they're not in the requested state